src/components/elements/Modal/component.js

```diff
@@ -11,14 +12,19 @@ import { Icon, ICON_NAMES } from 'elements/Icon';
 // eslint-disable-next-line jsdoc/require-jsdoc
 export const Component = ({
   children,
+  closeIcon,
+  hasRoundedCorners,
   isFullscreen,
   isOpen,
   onClose,
   size,
   trigger,
 }) => (
   <Modal
-    closeIcon={<Icon name={ICON_NAMES.CLOSE} />}
+    className={getClassNames({
+      'has-rounded-corners': hasRoundedCorners,
+    })}
+    closeIcon={closeIcon}
```

Comment:
```suggestion
    closeIcon={hasCloseIcon && <Icon name={ICON_NAMES.CLOSE} />}
```

Then we expose a boolean prop `hasCloseIcon`